### PR TITLE
fix for rpc timeout error when doing config store operations on agents that are not running

### PR DESCRIPTION
**Pull request by schandrika**
fix for rpc timeout error when doing config store operations on agents that are not running. 
jira-484

### CODE REVIEW

**Comment by laroque** [comment]
seems to be working as expected on RPi 4:

With platform running, agent off, config entry existing:
```
$ vctl config list platform.driver
devices/fake

$ vctl status
  AGENT                    IDENTITY        TAG             STATUS          HEALTH
c platform_driveragent-4.0 platform.driver platform_driver
```

remove config entry:
```
$ vctl config delete --all platform.driver
$ echo $?
0
$ vctl config list platform.driver
```

add config entry:
```
$ vctl config store platform.driver devices/fake examples/configurations/drivers/fake.config
$ echo $?
0
```

and list them again:
```
$ vctl config list platform.driver
devices/fake
```

**Review by laroque — APPROVED** [review]

File: volttron/platform/store.py [hunk 1]
```diff
@@ -153,6 +153,15 @@ def _setup(self, sender, **kwargs):
                                           "name_map": name_map,
                                           "lock": Semaphore()}
 
+    @Core.receiver('onstart')
+    def _onstart(self, sender, **kwargs):
+        try:
+            _log.info("Call peerlist() once so that we can access the peer_list variable after this")
```
Comment (craig8) [inline volttron/platform/store.py:159]: Please remove or make lower logging.

File: volttron/platform/store.py [hunk 2]
```diff
@@ -184,18 +193,21 @@ def manage_delete_store(self, identity):
...
-                                  trigger_callback=True).get(timeout=UPDATE_TIMEOUT)
-            except errors.Unreachable:
-                _log.debug("Agent {} not currently running. Configuration update not sent.".format(identity))
-            except RemoteError as e:
-                _log.error("Agent {} failure when all configurations: {}".format(identity, e))
-            except MethodNotFound as e:
-                _log.error(
-                    "Agent {} failure when deleting configuration store: {}".format(identity, e))
+        if identity not in self.vip.peerlist.peers_list:
+            _log.info("Agent is not currently running. Configuration updates not sent")
```
Comment (craig8) [inline volttron/platform/store.py:197]: I don't think this should be logged.  I think it should just not be sent.

File: volttron/platform/store.py [hunk 3]
```diff
@@ -293,20 +305,22 @@ def get_configs(self):
...
-            except errors.Unreachable:
-                _log.debug("Agent {} not currently running. Configuration update not sent.".format(identity))
-            except RemoteError as e:
-                _log.error("Agent {} failure when performing initial update: {}".format(identity, e))
-            except MethodNotFound as e:
-                _log.error(
-                    "Agent {} failure when performing initial update: {}".format(identity, e))
-            except errors.VIPError as e:
-                _log.error("VIP Error sending initial agent configuration: {}".format(e))
+        if identity not in self.vip.peerlist.peers_list:
```
Comment (craig8) [inline volttron/platform/store.py:308]: I don't think this should be logged. I think it should just not be sent.

File: volttron/platform/store.py [hunk 4]
```diff
@@ -348,16 +362,19 @@ def delete(self, identity, config_name, trigger_callback=False, send_update=True
...
-                try:
-                    self.vip.rpc.call(identity, "config.update", "DELETE", config_name, trigger_callback=trigger_callback).get(timeout=UPDATE_TIMEOUT)
-                except errors.Unreachable:
-                    _log.debug("Agent {} not currently running. Configuration update not sent.".format(identity))
-                except RemoteError as e:
-                    _log.error("Agent {} failure when deleting configuration {}: {}".format(identity, config_name, e))
-                except MethodNotFound as e:
-                    _log.error(
-                        "Agent {} failure when adding/updating configuration {}: {}".format(identity, config_name, e))
+            if identity not in self.vip.peerlist.peers_list:
```
Comment (craig8) [inline volttron/platform/store.py:365]: I don't think this should be logged. I think it should just not be sent.

File: volttron/platform/store.py [hunk 5]
```diff
@@ -429,17 +446,20 @@ def _add_config_to_store(self, identity, config_name, raw, parsed,
...
-                    _log.error("Agent {} failure when adding/updating configuration {}: {}".format(identity, config_name, e))
-                except MethodNotFound as e:
-                    _log.error(
-                        "Agent {} failure when adding/updating configuration {}: {}".format(identity, config_name, e))
-                except gevent.timeout.Timeout:
-                    _log.error("Config update to agent {} timed out after {} seconds".format(identity, UPDATE_TIMEOUT))
-                except Exception as e:
-                    _log.error("Unknown error sending update to agent identity {}.: {}".format(identity, e))
+            if identity not in self.vip.peerlist.peers_list:
+                _log.info("Agent is not currently running. Configuration updates not sent")
```
Comment (craig8) [inline volttron/platform/store.py:450]: I don't think this should be logged. I think it should just not be sent.

**Review by craig8 — CHANGES_REQUESTED** [review]
I wonder if there was a way to wrap these in a function call..

**Comment by schandrika** [comment]
Fixed review comments. @laroque Could you please do a quick test again. Thank you

**Review by craig8 — APPROVED** [review]

**Comment by laroque** [comment]
@schandrika same test as above, same behavior observed: no errors, no timeouts, commands all return are responsive speeds.

**Pull request merged**